skhep/dataset/numpydataset.py

```diff
@@ -115,12 +132,23 @@ def isdictof1d(data):
     def dicttoarray(self):
         """Convert a dictionnary into a structured array."""
         if self.isdictof1d(self.data ) and not self.isrecarray(self.data ):
-            dtypes = {'names': list(self.data.keys()), 'formats': [numpy.dtype(self.data [k].dtype) for k in self.data .keys()]}
+            if sys.version_info[0] > 2 and any(isinstance(k, bytes) for k in self.data.keys()):
```

Comment:
Worth adding comments/explanations when there is something especially done for Py2 or Py3 only.